interop/src/main.rs

```diff
@@ -176,6 +182,50 @@ fn run(log: Logger, options: Opt) -> Result<()> {
         }
     }
 
+    let mut h3 = false;
+    let result = runtime.block_on(
+        endpoint
+            .connect_with(&client_config, &remote, &options.host)?
+            .map_err(|e| format_err!("failed to connect: {}", e))
+            .and_then(|conn| {
+                assert!(state.lock().unwrap().saw_cert);
+                handshake = true;
+                let conn = conn.connection;
+                let control_stream = conn.open_uni();
+                let control_fut = control_stream
+                    .map_err(|e| format_err!("failed to open control stream: {}", e))
+                    .and_then(move |stream| {
+                        let mut buf = BytesMut::new();
+                        buf.put(0x43u8);
```

Comment:
What is this for?

Reply:
The control stream is useful to send one's endpoint settings https://quicwg.org/base-drafts/draft-ietf-quic-http.html#rfc.section.3.2.1. This is not mandatory for this code to work but I thought I would leave it as is to validate that the `SETTINGS FRAME` is correctly encoded.

Normally it shouldn't be chained with the request, but executed in parallel. This is something I plan to implement in future HTTP lifecycle module. Moreover the implementation should receive one `SETTINGS FRAME`  from the server and change it's QPACK encoder behavior accordingly. Which will involve shared mutable state.

Reply:
Does it need to involve a magic number hardcoded into the interop tool?

Reply:
re: shared mutable state: it's acceptable to have `Arc<Mutex<State>>` (perhaps hidden in a helper type that the user constructs per-stream?) for cases like this, IMO.

Reply:
Yeah, I was also mostly responding to the magic number, which seems a bit awkward and hard to understand.

Reply:
Ah yes, it's not very clean indeed...

Reply:
fixed in 51521a0